### PR TITLE
enhance Typescript definitions

### PR DESCRIPTION
**Pull request by robdg80**
- added definitions for scheduling
- fixed missing scheduling options on on/subscribe function
- enhance documentation of findings in node-schedule

Sorry for mistakes, I'm beginner in Javascript/Typescript/GitHub.

### CODE REVIEW

**Comment by AlCalzone** [comment]
I'll take a look at it later or tomorrow. Pls don't merge until then @GermanBluefox ;)

File: lib/javascript.d.ts [hunk 1]
```diff
@@ -342,9 +342,104 @@ declare global {
...
+			start?: Date | string | number;
+			/**
+			 * set an end time for schedule
+			 * a Data object or a dateString resp a number in milliseconds which can create a Date object
+			 */
+			end?: Date | string | number;
+			/**
+			 * timezone which should be used
+			 */
+			tz?: any;
```
Comment (AlCalzone) [inline lib/javascript.d.ts]: tz is a string => https://github.com/moment/moment-timezone

File: lib/javascript.d.ts [hunk 2]
```diff
@@ -342,9 +342,104 @@ declare global {
...
+			second?: number | number[] | string | string[];
+
+			/**
+			 * Year.
+			 */
+			year?: number | number[] | string | string[];
+			/**
+			 * timezone which should be used
+			 */
+			tz?: any;
```
Comment (AlCalzone) [inline lib/javascript.d.ts]: tz is a string => https://github.com/moment/moment-timezone

File: lib/javascript.d.ts (hunk 1 above)
Comment (robdg80) [reply lib/javascript.d.ts]: Thank You.
I have searched the right Type, but unfortunately, I was not successful. I was kinda wrong looking in GitHub.
Sorry, I'm not so familar with Javascript.

**Comment by robdg80** [comment]
I have hopefully corrected that correctly.

**Comment by AlCalzone** [comment]
I proposed my changes as a PR on his fork. Once that is in, we can merge IMO.

**Pull request merged**